Control/Applicative/Combinators/NonEmpty.hs

```diff
@@ -0,0 +1,49 @@
+-- |
+-- This module provides non-empty list variants of some of the functions in
+-- "Control.Applicative.Combinators" and re-exports `some1` from
+-- "Data.List.NonEmpty".
+
+module Control.Applicative.Combinators.NonEmpty
+  ( endBy1
+  , sepBy1
+  , sepEndBy1
+  , some1
+  , someTill
+  )
+where
+
+import Control.Applicative
+import Data.List.NonEmpty
```

Comment:
You need to add something like:

```
  if !impl(ghc >= 8.0)
    build-depends:    semigroups   == 0.18.*
```

To the Cabal file. Older `base` versions do not come with `Data.List.NonEmpty`.

----

Also please have it like this:

```
import Data.List.NonEmpty (NonEmpty (..))
import qualified Data.List.NonEmpty as NE
```

This way it'll be a bit clearer which `fromList` is meant. Sorry for nitpicking.